doc/source/api.rst

```diff
@@ -1,10 +1,259 @@
+``num_gpus`` argument. Similarly, you can also specify the number of CPUs with
+``num_cpus``.
+
+.. code-block:: python
+
+  ray.init(num_cpus=20, num_gpus=2)
+
+By default, Ray will use ``psutil.cpu_count()`` to determine the number of CPUs,
+and by default the number of GPUs will be zero.
+
+Connecting to an existing cluster
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once a Ray cluster has been started, the only thing you need in order to connect
+to it is the address of the Redis server in the cluster. In this case, your
+script will not start up or shut down any processes. The cluster and all of its
+processes may be shared between multiple scripts and multiple users. To do this,
+you simply need to know the address of the cluster's Redis server. This can be
+done with a command like the following.
+
+   .. code-block:: python
+
+     ray.init(redis_address="12.345.67.89:6379")
+
+In this case, you cannot specify ``num_cpus`` or ``num_gpus`` in ``ray.init``
+because that information is passed into the cluster when the cluster is started,
+not when your script is started.
+
+View the instructions for how to `start a Ray cluster`_ on multiple nodes.
+
+.. _`start a Ray cluster`: http://ray.readthedocs.io/en/latest/using-ray-on-a-cluster.html
+
+.. autofunction:: ray.init
+
+Defining remote functions
+-------------------------
+
+Remote functions are used to create tasks. To define a remote function, the
+``@ray.remote`` decorator is placed over the function definition.
+
+The function can then be invoked with ``f.remote``. Invoking the function
+creates a **task** which will be scheduled on and executed by some worker
+process in the Ray cluster. The call will return an **object ID** (essentially a
+future) representing the eventual return value of the task.
+
+When a remote function returns, its outputs will be serialized into a string of
+bytes and stored in an object store (whichever object store the worker that
+executed the task is connected to).
+
+Note that arguments to remote functions can be values or object IDs.
+
+.. code-block:: python
+
+  @ray.remote
+  def f(x):
+    return x + 1
+
+  x_id = f.remote(0)
+  ray.get(x_id)  # 1
+
+  y_id = f.remote(x_id)
+  ray.get(y_id)  # 2
+
+If you want a remote function to return multiple object IDs, you can do that by
+passing the ``num_return_vals`` argument into the remote decorator.
+
+.. code-block:: python
+
+  @ray.remote(num_return_vals=2)
+  def f():
+    return 1, 2
+
+  x_id, y_id = f.remote()
+  ray.get(x_id)  # 1
+  ray.get(y_id)  # 2
+
 .. autofunction:: ray.remote
+
+Getting values from object IDs
+------------------------------
+
+Object IDs can be converted into objects by calling ``ray.get`` on the object
+ID. When a driver or worker process calls ``ray.get`` on an object ID, the
+corresponding object will be transferred to the driver or worker's local object
+store (from whichever object store it is in). It will then be deserialized and
+turned into a regular Python object.
```

Comment:
I think the implications here aren't super clear. The two I can think of are:
- Python objects need to be *copied* into the worker's memory and deserialized there
- Mutations done to the deserialized object will not be reflected in the object store's copy